dockerspawner/systemuserspawner.py

```diff
@@ -101,6 +115,7 @@ def get_env(self):
             NB_USER=self.user.name,
             USER_ID=self.user_id, # deprecated
             NB_UID=self.user_id,
+            NB_GID=self.group_id,
```

Comment:
This probably doesn't want to be set `if self.group_id < 0`

Reply:
I added a check for that, I hope it's what you wanted.

https://github.com/jupyterhub/dockerspawner/pull/304/commits/10183dcab732b1cc2c1c999ccbafe29cf4aeaf79